### DISCUSSION TITLE
What is the best & easy way to write tests to know that an observable emits N values in order? I would like to do it like in RxDart

### DISCUSSION THREAD

**Discussion by MiniSuperDev** [ReactiveX/rxjs#7101] [Q&A] [answered]
Hello, I am migrating from `RxDart` to `RxJs` with typescript, I would like to know what is the best and easy way to write tests to know that an observable emits N values in order.

I see there is [marble-testing](https://rxjs.dev/guide/testing/marble-testing) but I'm looking for something more simple or similar to how it was done in dart.

For example in RxDart, it was very intuitive using expect with `emitsInOrder` or any of the [stream matchers](https://pub.dev/packages/test#stream-matchers), but I don't see anything similar in typescript.

https://github.com/ReactiveX/rxdart/blob/master/test/streams/sequence_equals_test.dart#L7-L12
https://github.com/ReactiveX/rxdart/blob/master/test/streams/combine_latest_test.dart#L21-L36

Thank you

**Comment by kwonoj** [comment, marked as answer]
Marble testing is not a mandatory requirement, you could grab values via subscribing test scheduler and compare values. There are 3rd party testing framework like rx-sandbox (https://github.com/kwonoj/rx-sandbox#getting-values-from-observable) as well allows to use plain testmessage assertion.

disclaimer: I wrote rx-sandbox.
